README.md

```diff
@@ -21,14 +19,42 @@ This project framework provides the following features:
 * ...
  
 ### Architecture Diagram
+
 Include a diagram describing the application
 
 ### Demo Video (optional)
+
 (Embed demo video here)
  
 ## Getting Started
- 
-### Prerequisites
+
+You have a few options for getting started with this template.
+The quickest way to get started is GitHub Codespaces, since it will setup all the tools for you, but you can also [set it up locally](#local-environment).
+
+### GitHub Codespaces
+
+You can run this template virtually by using GitHub Codespaces. The button will open a web-based VS Code instance in your browser:
+
+1. Open the template (this may take several minutes)
+    [![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](placeholder)
+2. Open a terminal window
+3. [any other steps needed for your template]
+4. Run `azd up`
+
+### VS Code Dev Containers
+
+A related option is VS Code Dev Containers, which will open the project in your local VS Code using the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers):
+
+1. Start Docker Desktop (install it if not already installed)
+2. Open the project:
+    [![Open in Dev Containers](https://img.shields.io/static/v1?style=for-the-badge&label=Dev%20Containers&message=Open&color=blue&logo=visualstudiocode)](placeholder)
+3. In the VS Code window that opens, once the project files show up (this may take several minutes), open a terminal window.
+4. [any other steps needed for your template]
+5. Run `azd up`
```

Comment:
For `azd up` here and above, do we want to put it in a code block instead? Then its easy to copy/paste. (I know its a short command, so we vary in what we do - just for your consideration)

Reply:
For example:

```
3. Sign in to your Azure account:

    ```shell
    azd auth login --use-device-code
    ```

4. Provision the resources and deploy the code:

    ```shell
    azd up
    ```
```